backend-rust/src/indexer.rs

```diff
@@ -2849,13 +2912,16 @@ impl PreparedContractUpdates {
         tx: &mut sqlx::Transaction<'static, sqlx::Postgres>,
         transaction_index: i64,
     ) -> anyhow::Result<()> {
-        for elm in &self.trace_elements {
-            elm.save(tx, transaction_index).await?;
+        for (i, elm) in self.trace_elements.iter().enumerate() {
+            elm.save(tx, transaction_index).await.with_context(|| {
+                format!("Failed processing contract update trace element with index {}", i)
```

Comment:
I don't think we print the trace element in case of an error anywhere, so would be good to add.

```suggestion
                format!("Failed processing contract update trace element {:?} with index {}", elm, i)
```